### PR TITLE
Add helper event name builder

### PR DESCRIPTION
**Pull request by nozim**
Closes: #437 

## Description

______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-go-sdk/blob/master/CONTRIBUTING.md#styleguides).
- [ ] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

**Comment by sideninja** [comment]
@chasefleming mind reviewing?

**Comment by PratikDhanave** [comment]
@nozim PR looks good to me one suggestion can we omit 'with' word' and make it as 
customType := flow.NewEvent().
	         EventName("Add").
		ContractName("EventDemo").
		Address(deployedContract.Address).
		Build()

**Comment by nozim** [comment]
@PratikDhanave you are most welcome to propose your own solution in separate PR :)

**Comment by PratikDhanave** [comment]
> nozim

sure will do thank you

File: examples/get_events/main.go [hunk 1]
```diff
@@ -45,7 +45,14 @@ func demo(deployedContract *flow.Account, runScriptTx *flow.Transaction) {
 	printEvents(result, err)
 
 	// Query for our custom event by type
-	customType := fmt.Sprintf("AC.%s.EventDemo.EventDemo.Add", deployedContract.Address.Hex())
+	//customType := fmt.Sprintf("AC.%s.EventDemo.EventDemo.Add", deployedContract.Address.Hex())
```
Comment (chasefleming) [inline examples/get_events/main.go:48]: Remove this please

File: event_name_factory_test.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+package flow
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestEventNameFactory(t *testing.T) {
+	assert.Equal(t, "A.7e60df042a9c0868.FlowToken.AccountCreated", NewEvent().
```
Comment (chasefleming) [inline event_name_factory_test.go:28]: I noticed you took a slightly different approach with the chaining than on the original issue. Can you explain why you chose this approach over the other?

**Review by chasefleming — CHANGES_REQUESTED** [review]
Excited for this utility! Will be useful. One question and a comment. Thanks for the contribution!

File: event_name_factory.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	f.eventName = event
+	return f
+}
+
+func (f eventNameFactory) Build() string {
+	return fmt.Sprintf("A.%s.%s.%s", f.address, f.contractName, f.eventName)
+}
+
+// NewEvent helper function for constructing event names
+func NewEvent() eventNameFactory {
```
Comment (nvdtf) [inline event_name_factory.go:56]:
```suggestion
func NewEventTypeFactory() eventTypeFactory {
```
Also `eventNameFactory` -> `eventTypeFactory` (you can also move this code into `event.go`). 

I suggest making the name more explicit to keep in line with cadence terminology and avoid confusion with `cadence.Event`. Go SDK already calls this `Type` in `event.go`. The general `Event` type is expected to have fields and values.

https://github.com/onflow/cadence/blob/52549529e96b6094bb726e5e50d63f34e2eb292b/types.go#L1595-L1601

https://github.com/onflow/flow-go-sdk/blob/3f7414701e1aab43b272f791b7385ae9e329aca8/event.go#L42

cc @sideninja it was part of the original design would like to know your thoughts as well

File: event_name_factory_test.go (hunk 2 above)
Comment (chasefleming) [reply event_name_factory_test.go:28]: This pattern actually does seem to fit the patterns in the rest of the SDK. Thoughts on changing `.Build` to `.String` for more flexibility with the event struct later though?

**Review by nvdtf — CHANGES_REQUESTED** [review]

**Comment by nozim** [comment]
@nvdtf @chasefleming thanks for feedback, will fix them shortly.

File: event_name_factory.go (hunk 3 above)
Comment (nozim) [reply event_name_factory.go]: @nvdtf  amended here https://github.com/onflow/flow-go-sdk/pull/449/commits/155a11e76499d84ef5487b8d1ca7cd29d57cf428

File: event_name_factory_test.go (hunk 2 above)
Comment (nozim) [reply event_name_factory_test.go]: updated here https://github.com/onflow/flow-go-sdk/pull/449/commits/155a11e76499d84ef5487b8d1ca7cd29d57cf428.

@chasefleming I'd be happy to propose the original design if you fancy ;)

File: examples/get_events/main.go (hunk 1 above)
Comment (nozim) [reply examples/get_events/main.go]: removed here https://github.com/onflow/flow-go-sdk/pull/449/commits/155a11e76499d84ef5487b8d1ca7cd29d57cf428

File: event_name_factory.go (hunk 3 above)
Comment (nvdtf) [reply event_name_factory.go]: Thanks! Some leftovers:
* filename`event_name_factory.go` should be changed. I suggest moving this code into `events.go` or if you want to keep it separate, rename the file to `event_type_factory.go`
* `event_name_factory_test.go` -> `event_type_factory_test.go`
* Test function name ` TestEventNameFactory` -> ` TestEventTypeFactory`

Comment (nozim) [reply event_name_factory.go]: @nvdtf fixed here https://github.com/onflow/flow-go-sdk/pull/449/commits/f99825351b3ebea44e6d460369939f32c572819c.

Thank you very much for pointing out.

**Review by nvdtf — APPROVED** [review]

**Review by chasefleming — APPROVED** [review]

**Pull request merged**